### PR TITLE
view list of items

### PR DESCRIPTION
**Pull request by JeffNewcomer**

### CODE REVIEW

**Comment by JeffNewcomer** [comment]
screen shot 2018-06-26 at 4 20 31 pm

File: main.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+  var $row1 = document.createElement('div')
+  $row1.classList.add('row')
+  $grid.appendChild($row1)
+
+  for (var i = 0; i < allItems.length; i++) {
+    var $cardDiv = document.createElement('div')
+    $cardDiv.classList.add('col-md-3')
+    $row1.appendChild($cardDiv)
+    var $newCard = createCard(app.catalog.items[i])
```
Comment (thebearingedge) [inline main.js]: Instead of reaching out to `app.catalog.items`, you should be able to use your `allItems` parameter.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
> Define a function that renders the entire app state and inserts it into the view.

I can only see two function definitions. There should be three when this is done.

File: main.js (hunk 1 above)
Comment (JeffNewcomer) [reply main.js]: Done.

Comment (JeffNewcomer) [reply main.js]: Still working on the 3rd function.

**Comment by thebearingedge** [comment]
Please re-open this when it's ready.

**Comment by JeffNewcomer** [comment]
Created a 3rd function to render the grid and header to the view and changed app.catalog.items to the parameter AllItems within the createGridAndHeading function.
screen shot 2018-06-26 at 4 20 31 pm

File: main.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    $row1.appendChild($cardDiv)
+    var $newCard = createCard(allItems[i])
+    $cardDiv.appendChild($newCard)
+  }
+  return $grid
+}
+// Define a function that renders the entire app state and inserts it into the view.
+function renderGridAndHeader() {
+  var i = createGridAndHeading(app.catalog.items)
+  document.body.appendChild(i)
```
Comment (thebearingedge) [inline main.js]: Instead of appending this to the `body`, there is a specific place to insert the rendered catalog in `index.html`.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Awesome, one last change.

File: main.js (hunk 2 above)
Comment (JeffNewcomer) [reply main.js]: `function renderGridAndHeader() {
  var i = createGridAndHeading(app.catalog.items)
  var $catalog = document.querySelector("div[data-view='catalog']")
  $catalog.appendChild(i)
}
renderGridAndHeader()`

**Pull request merged**